src/configuration.js

```diff
@@ -36,26 +39,38 @@ export class Configuration {
     // TODO Cannot have both except and only -- raise in this case
     // TODO validate that all rules passed to only/except exist.
 
-    var rules;
+    var rules = defaultRules;
 
     if (this.options.only.length > 0) {
-      rules = defaultRules.filter(rule => {
-        return this.options.only.map(toUpperCamelCase).indexOf(rule.name) >= 0;
-      });
+      rules = filterRules(this.options.only);
     } else if (this.options.except.length > 0) {
       rules = defaultRules.filter(rule => {
         return (
           this.options.except.map(toUpperCamelCase).indexOf(rule.name) == -1
         );
       });
     } else {
-      rules = defaultRules;
+      const directory = this.options.searchDirectory || process.cwd();
+      const cosmic = cosmiconfig('graphql-schema-linter', {
```

Comment:
Along the same line of thinking, we might want to move this to a `func loadOptionsFromConfig()` that gets called in the `constructor` otherwise we will have this same boilerplate in `getSchema`.

We could also do something like this in the `constructor`:

```js
constructor(options, stdinFd, configSearchDirectory) {
  const configOptions = this.loadOptionsFromConfig(configSearchDirectory);
  this.options = Object.assign(configOptions, options);
}
```

This way CLI options take precedence over config. It also enables us to easily add support for [`.graphqlconfig`](https://github.com/graphcool/graphql-config) in the future.

I think the beauty of doing this in the `constructor` is that none of the getters need to be changed.